buildSrc/src/main/java/com/google/firebase/gradle/plugins/ReleaseGenerator.kt

```diff
@@ -261,5 +261,25 @@ abstract class ReleaseGenerator : DefaultTask() {
   private fun getRelativeDir(project: Project) = project.path.substring(1).replace(':', '/')
 }
 
+abstract class ReleaseReportGenerator : DefaultTask() {
+
+  @get:InputFiles abstract val releaseReportJsonFile: RegularFileProperty
+
+  @get:OutputFile abstract val releaseReportMdFile: RegularFileProperty
+
+  @TaskAction
+  @Throws(Exception::class)
+  fun generateReleaseReport() {
+    val rootDir = project.rootDir
+
+    val releaseReport =
+      ReleaseReport.fromFile(
+        releaseReportJsonFile.asFileIfExistsOrNull()
+          ?: throw RuntimeException("Missing release json file")
+      )
+    releaseReportMdFile.get().asFile.writeText(releaseReport.toString())
+  }
+}
+
 fun RegularFileProperty.asFileIfExistsOrNull(): File? =
```

Comment:
This could actually be reduced to

```kotlin
fun RegularFileProperty.asFileIfExistsOrNull(): File? =
  orNull?.asFile?.takeIf { it.exists() }
```